hub-saml/src/test/java/uk/gov/ida/saml/hub/api/HubTransformersFactoryTest.java

```diff
@@ -47,57 +53,61 @@ public void setUp() {
             .withLevelsOfAssurance(Collections.singletonList(AuthnContext.LEVEL_3))
             .buildFromHub();
 
-        //When
         String apply = eidasTransformer.apply(idaAuthnRequestFromHub);
 
-        //Then the Authn Request string is returned
         Assert.assertNotNull(apply);
 
-        //And the Authn Request does not contain a KeyInfo section for Verify UK
         AuthnRequest authnReq = stringtoOpenSamlObjectTransformer.apply(apply);
         Assert.assertNotNull(authnReq);
         Assert.assertNull("The Authn Request does not contain a KeyInfo section for Verify UK", authnReq.getSignature().getKeyInfo());
     }
 
     @Test
     public void getEidasAuthnRequestFromHubToStringTransformer_testContainsKeyInfo() throws Exception {
-        //Given
-        SignatureAlgorithm signatureAlgorithm = new SignatureRSASHA256();
-        DigestAlgorithm digestAlgorithm = new DigestSHA256();
         Function<EidasAuthnRequestFromHub, String> eidasTransformer = new HubTransformersFactory().getEidasAuthnRequestFromHubToStringTransformer(
-            getKeyStore(),
+            getKeyStore(true),
             signatureAlgorithm,
             digestAlgorithm
         );
         EidasAuthnRequestFromHub eidasAuthnRequestFromHub = EidasAuthnRequestBuilder.anEidasAuthnRequest()
             .withLevelsOfAssurance(Collections.singletonList(AuthnContext.LEVEL_2))
             .buildFromHub();
 
-        //When
         String apply = eidasTransformer.apply(eidasAuthnRequestFromHub);
 
-        //Then the Authn Request string is returned
         Assert.assertNotNull(apply);
 
-        //And the Authn Request contains a KeyInfo section for eIDAS
         AuthnRequest authnReq = stringtoOpenSamlObjectTransformer.apply(apply);
         Assert.assertNotNull(authnReq);
         Assert.assertNotNull("The Authn Request contains a KeyInfo section for eIDAS", authnReq.getSignature().getKeyInfo());
     }
 
-    private static IdaKeyStore getKeyStore() throws Base64DecodingException {
+    @Test
+    public void shouldThrowExceptionWhenKeyInfoIsRequiredButSigningCertIsNotPresent() throws Base64DecodingException {
+        expectedException.expectMessage("Unable to generate key info without a signing certificate");
+
+        Function<EidasAuthnRequestFromHub, String> eidasTransformer = new HubTransformersFactory().getEidasAuthnRequestFromHubToStringTransformer(
+                getKeyStore(false),
+                signatureAlgorithm,
+                digestAlgorithm
+        );
+        EidasAuthnRequestFromHub eidasAuthnRequestFromHub = EidasAuthnRequestBuilder.anEidasAuthnRequest()
+                .withLevelsOfAssurance(Collections.singletonList(AuthnContext.LEVEL_2))
+                .buildFromHub();
+
+        eidasTransformer.apply(eidasAuthnRequestFromHub);
+    }
+
+    private static IdaKeyStore getKeyStore(boolean includeSigningCert) throws Base64DecodingException {
```

Comment:
Bit opaque having the signing cert inclusion hidden in a boolean parameter but should do for now.